app/jobs/regular/create_linked_topic.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Jobs
+  class CreateLinkedTopic < ::Jobs::Base
+
+    def execute(args)
+      reference_post = Post.find_by(id: args[:post_id])
+      return unless reference_post.present?
+      parent_topic = reference_post.topic
+      return unless parent_topic.present?
+      parent_topic_id = parent_topic.id
+      parent_title = parent_topic.title
+      title_suffix_locale = I18n.t('create_linked_topic.title_suffix')
+
+      linked_topic_record = parent_topic.linked_topic
+      if linked_topic_record.present?
+        raw_title = parent_title.gsub(/\(#{title_suffix_locale} (\d+)\)$/i, "").strip
+        original_topic_id = linked_topic_record.original_topic_id
+        sequence = linked_topic_record.sequence + 1
+      else
+        raw_title = parent_title
+
+        # update parent topic title to append title_suffix_locale
+        parent_title = "#{parent_title} (#{title_suffix_locale} 1)"
+        parent_topic.title = parent_title
+        parent_topic.save!
+
+        # create linked topic record
+        original_topic_id = parent_topic_id
+        LinkedTopic.create!(topic_id: parent_topic_id, original_topic_id: original_topic_id, sequence: 1)
+        sequence = 2
+      end
+
+      # fetch previous topic titles
+      previous_topics = ""
+      linked_topic_ids = LinkedTopic.where(original_topic_id: original_topic_id).pluck(:topic_id)
+      Topic.where(id: linked_topic_ids).order(:id).each do |topic|
+        previous_topics += "- [#{topic.title}](#{topic.url})\n"
+      end
+
+      # create new topic
+      new_topic_title = "#{raw_title} (#{title_suffix_locale} #{sequence})"
+      new_topic_raw = I18n.t('create_linked_topic.post_raw', parent_title: "[#{parent_title}](#{reference_post.full_url})", previous_topics: previous_topics)
```

Comment:
We should run this by @gschlager who knows best about translation keys. What do you think of this approach of building a list and sending it over to one key? (Also, it's probably worth reading the above part that creates the topic titles.)

Reply:
Seems fine as it just concatenates multiple Markdown strings.